media/js/firefox/concerts.js

```diff
@@ -1,6 +1,173 @@
+
+            // Set a cookie so we don't have to query location on repeated page loads.
+            ConcertPage.setCookie(country);
+
+            // Update page content based on location.
+            ConcertPage.updatePageContent();
+        }
+    };
+
+    ConcertPage.updatePageContent = function() {
+        if (ConcertPage.shouldShowConcert() === 'false') {
+            ConcertPage.showExcludedContent();
+        } else {
+            ConcertPage.showConcertContent();
+        }
+    };
+
+    ConcertPage.showConcertContent = function() {
+        // Add styling hook for us-specific CSS.
+        document.body.classList.add('state-in-us');
+    };
+
+    ConcertPage.showExcludedContent = function() {
+        // Add styling hook for excluded-specific CSS.
+        document.body.classList.add('state-not-us');
+    };
+
+    ConcertPage.shouldShowConcert = function() {
+        // Is user in the US?
+        return ConcertPage.verifyLocation(ConcertPage.getCookie(ConcertPage.COOKIE_ID));
+    };
+
+    ConcertPage.cookieExpiresDate = function(date) {
+        var d = date || new Date();
+        d.setTime(d.getTime() + (ConcertPage.COOKIE_EXPIRATION_DAYS * 24 * 60 * 60 * 1000));
+        return d.toUTCString();
+    };
+
+    ConcertPage.setCookie = function(country) {
+        Mozilla.Cookies.setItem(ConcertPage.COOKIE_ID, country, ConcertPage.cookieExpiresDate());
+    };
+
+    ConcertPage.getCookie = function(id) {
+        return Mozilla.Cookies.getItem(id);
+    };
+
+    ConcertPage.hasCookie = function() {
+        return Mozilla.Cookies.hasItem(ConcertPage.COOKIE_ID);
+    };
+
+    ConcertPage.init = function() {
+        ConcertPage.passThroughParams();
+
+        var cookiesEnabled = typeof Mozilla.Cookies !== 'undefined' || Mozilla.Cookies.enabled();
+        var override = ConcertPage.hasGeoOverride();
+
+        // only show excluded content if cookies are enabled (let LiveNation deal with it)
+        if (cookiesEnabled) {
+            // if override URL is used, skip doing anything with cookies & show the expected content.
+            if (override) {
+                if (ConcertPage.verifyLocation(override)) {
+                    ConcertPage.showExcludedContent();
+                } else {
+                    ConcertPage.showConcertContent();
+                }
+            } else {
+                // if user already has a cookie, use that data and update page content straight away.
+                if (ConcertPage.hasCookie()) {
+                    ConcertPage.updatePageContent();
+                }
+                // else make a remote call to query location.
+                else {
+                    ConcertPage.getLocation();
+                }
+            }
+        } else {
+            ConcertPage.showConcertContent();
+        }
+    };
+    window.Mozilla.ConcertPage = ConcertPage;
+
+
     var stateStorageKey = 'fxaOauthState';
     var verifiedStorageKey = 'fxaOauthVerified';
     var cookieDays = 1;
```

Comment:
Why 1? That seems short.

Reply:
Ask jpetto, he wrote this part ;) But yeah, I think I'll bump that up... maybe a week? Two weeks?